public/js/app/ladderWeek.js

```diff
@@ -0,0 +1,124 @@
+  $("#navigation-last-"+category).addClass("disabled");
+  if (currentPage[category] < lastPage[category]){
+      $("#navigation-next-"+category).removeClass("disabled");
+      $("#navigation-last-"+category).removeClass("disabled");
+  }
+  if (currentPage[category] > 1){
+      $("#navigation-first-"+category).removeClass("disabled");
+      $("#navigation-previous-"+category).removeClass("disabled");
+  }
+  const elementId = "players"+category;
+  renderPage(category, document.getElementById(elementId), id); 
+};
+
+var renderPage = function (category, element, playerId) {
+    
+  const list = categories[category].data;
+
+  removeAllChildElements(element);
+  
+  for(var i = (currentPage[category]-1)*pageSize; i < Math.min(pageSize*currentPage[category], list.length); i++) {
+    var player = list[i];
+    var tr = document.createElement("tr");
+    
+    tr.setAttribute("id", "tr" + i);
+    element.appendChild(tr);
+    
+    // #
+    var rank = document.createElement("td");
+    tr.appendChild(rank);
+    rank.innerHTML = (i+1);
+    
+    // Player
+    var name = document.createElement("td");
+    tr.appendChild(name);
+    name.innerHTML = player.name;
+    
+    // Rating
+    var rating = document.createElement("td");
+    tr.appendChild(rating);
+    rating.innerHTML = player.rating;
+    
+    // Score
+    var points = document.createElement("td");
+    tr.appendChild(points);
+    points.innerHTML = player.points;
+    
+    // W/L/D
+    var wld = document.createElement("td");
+    tr.appendChild(wld);
+    wld.innerHTML = player.wld.w+"/"+player.wld.l+"/"+player.wld.d;
+    
+    
+    // Time played
+    var timeplayed = document.createElement("td");
+    tr.appendChild(timeplayed);
+    timeplayed.innerHTML = (moment(player.secondsPlayed*1000).hours()-1)+"h "+moment(player.secondsPlayed*1000).minutes()+"m "+moment(player.secondsPlayed*1000).seconds()+"s";
+  }
+};
+
+/* Utilities */
+
+var removeAllChildElements = function (element) {
+  while (element.firstChild) {
+    element.removeChild(element.firstChild);
+  }
+};
+
+/* Page Onclick */
+
+for (let catName in categories){
+     $("#navigation-previous-"+catName).click( function() {
+      if (currentPage[catName] === 1) {
+        return;
+      }
+
+      currentPage[catName]--;
+      getPage(catName, currentPage[catName], pageSize);
+    });
+
+     $("#navigation-next-"+catName).click( function() {
+      if (currentPage[catName] === lastPage[catName]) {
+        return;
+      }
+
+      currentPage[catName]++;
+      getPage(catName, currentPage[catName], pageSize);
+    });
+    
+    $("#navigation-first-"+catName).click( function() {
+      currentPage[catName] = 1;
+      getPage(catName, currentPage[catName], pageSize);
+    });
+
+    $("#navigation-last-"+catName).click( function() {
+      currentPage[catName] = lastPage[catName];
+      getPage(catName, currentPage[catName], pageSize);
+    });
+}
```

Comment:
this for block has a lot of duplicate code

Reply:
it is original though, from the leaderboards.js. Will fix.